Add the dev's instruction for the issue #13 (#327)
JeffersonLab/clas12-timeline: doc/dev_notes.md

- often we will run one and not the other: `--focus-detectors` needs `mon` schema, whereas `--focus-physics` prefers high statistics

# Notes on Writing the Analysis Scripts

There are a few cases where a developer may want to make some updates to the script.

> We strongly encourage you to work on this in the different branches from `main'.

Reviewer: ```suggestion
We strongly encourage you to work on this in the different branches from `main` and send pull requests.
```

> To add more analysis scripts for step 2 (see [the Chefs' documentation wiki](https://clasweb.jlab.org/wiki/index.php/CLAS12_Chef_Documentation) for details), create corresponding scripts in https://github.com/JeffersonLab/clas12-timeline/tree/main/src/main/java/org/jlab/clas/timeline/analysis.

Reviewer: - No need to link Chef's wiki, this is for developers.
- It's better (and easier) to generate source code links like this:
```markdown
[`path/to/directory`](/path/to/directory)
```
- use backticks around the link name in the square brackets (so it looks nice)
- use a leading `/` in the link path in the round parentheses, as a sort of "absolute path", so it won't break if we move this `dev_notes.md` file

```suggestion
To add more analysis scripts for step 2, create corresponding scripts in [`src/main/java/org/jlab/clas/timeline/analysis`](/src/main/java/org/jlab/clas/timeline/analysis).
```
Same goes for the other links here.

The histogram (`H1F`) and graph (`GraphErrors`) objects should be included in the output `TDirectory` object. Not all, but most, timeline variables are fit results. We maintain the fitters in https://github.com/JeffersonLab/clas12-timeline/tree/main/src/main/java/org/jlab/clas/timeline/fitter .
To display the fit results on the clas12mon website, the javascript rendering the fit function needs the name of `F1D` to be "fit: " + the name of `H1F` (see /group/clas/www/clas12mon/html/timeline/index.js in the ifarm.)

> If the `H1F` objects do not exist in the monitoring output, the java codes in https://github.com/JeffersonLab/clas12-timeline/tree/main/src/main/java/org/jlab/clas/timeline/histograms need to be updated for step 1 (see [the Chefs' documentation wiki](https://clasweb.jlab.org/wiki/index.php/CLAS12_Chef_Documentation) for details).

Reviewer: ```suggestion
If the `H1F` objects do not exist in the monitoring output, the java codes in https://github.com/JeffersonLab/clas12-timeline/tree/main/src/main/java/org/jlab/clas/timeline/histograms need to be updated for step 1.
```
